### ISSUE TITLE
Clear up confusion w/ presence around belongs_to associations. (#1214) breaks existing behaviors.

### ISSUE THREAD

**Issue by kamillle** [thoughtbot/shoulda-matchers#1218]
`v4.1.0` includes a change `Clear up confusion w/ presence around belongs_to associations.` (#1214  ) . This breaks existing behaviors.

My application code is below.

```
class HogeForm
  include ActiveModel::Model
  include ActiveModel::Attributes

  attribute :user
  attribute :note, :string

  validates :user, presence: true
  validates :note, presence: true, if: %i[user]

  def save
    return false if invalid?

    # do something
  end
end
```

And test code is below.

```
RSpec.describe HogeForm do
  context 'Validations' do
    it { is_expected.to validate_presence_of(:user) }
    ...
    ...
    ...
  end
end
```

When execute `it { is_expected.to validate_presence_of(:user) }`, then user attribute is inserted `nil` until v4.0.1, but in v.4.1.0 user attribute is inserted `""` (empty string) and `validates :note, presence: true, if: %i[user]` is run unintentional.

Why did you determine to change the behavior? and why not restore the behavior?

**Comment by mcmire** [comment]
The presence validation calls `.present?` on your attribute to determine whether it's "present". Presence is defined as either an empty string or nil: https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/blank.rb#L6. Therefore, to truly test the presence validation, you need to see what happens if your attribute is an empty string. The matcher really should have been doing that all along, so that's why I added it.

That said, I can change it so that it only acts on an ActiveRecord model so that your tests pass again.

**Comment by mcmire** [comment]
@kamillle Looking back at this, I'm having trouble writing a failing test for this based on your example. Are your tests breaking currently, and if so, which ones?

**Comment by kamillle** [comment]
@mcmire oops...
Sorry for providing wrong code.
Correct code is below, plz check it.

```
### application code ###
class HogeForm
  include ActiveModel::Model
  include ActiveModel::Attributes

  attribute :user

  validates :user, presence: true
  validate :validate_something, if: %i[user]

  def save
    return false if invalid?

    # do something
  end

  private

  def validate_something
    return if user.email

    errors.add(:base, 'Error!!!')
  end
end

### test code ###
# frozen_string_literal: true

require 'rails_helper'

RSpec.describe HogeForm, type: :model do
  context 'Validations' do
    it { is_expected.to validate_presence_of(:user) }
  end
end
```

**Comment by mcmire** [comment]
@kamillle Hey, I just pushed up a branch, `fix-presence-validation-for-activemodel`, that should fix this issue. Can you temporarily switch to it in your Gemfile and see if makes your tests pass?

**Comment by kamillle** [comment]
@mcmire Sure. I'll check that!! Thank you!😍

**Comment by kamillle** [comment]
@mcmire The fix is completely working in my case!!! Appriciate your working.😆🙌🙌

**Comment by mcmire** [comment]
This fix is live in [v4.1.1](https://github.com/thoughtbot/shoulda-matchers/releases/tag/v4.1.1).
